sycl/plugins/unified_runtime/ur/adapters/cuda/event.hpp

```diff
@@ -158,22 +160,25 @@ ur_result_t forLatestEvents(const ur_event_handle_t *EventWaitList,
     return F(EventWaitList[0]);
   }
 
+  // TODO we should allocate this once instead of in every call
   std::vector<ur_event_handle_t> Events{EventWaitList,
                                         EventWaitList + NumEventsInWaitList};
   std::sort(Events.begin(), Events.end(),
             [](ur_event_handle_t Event0, ur_event_handle_t Event1) {
               // Tiered sort creating sublists of streams (smallest value first)
               // in which the corresponding events are sorted into a sequence of
               // newest first.
-              return Event0->getStream() < Event1->getStream() ||
-                     (Event0->getStream() == Event1->getStream() &&
-                      Event0->getEventID() > Event1->getEventID());
+              return !Event0 || !Event1 ||
```

Comment:
Is it a valid state for either of events to be null?

Reply:
In that case, should you not turn it to an assert, currently it is accepted.

Reply:
Yes good shout